src/pathfora.js

```diff
@@ -524,40 +559,52 @@
       // array of urlContains params is an or list, so if any are true evaluate valid to true
       if (phrases.indexOf('*') === -1) {
         phrases.forEach(function (phrase) {
+
           // legacy match allows for an array of strings, check if we are legacy or current object approach
           switch (typeof phrase) {
             case 'string':
-              if (url.indexOf(phrase) !== -1) {
-                valid = true;
+              var matchUrl = decodeURIComponent(phrase);
+
+              if (url.indexOf(matchUrl.split("?")[0]) !== -1) {
+                valid = core.compareQueries(queries, core.parseQueries(matchUrl)) && true;
```

Comment:
this one is doing a soft compare, whereas the other is looking for an exact match. might need to methods or a wrapper method or something to handle these better. maybe just a param to do the last check for length?